### PR TITLE
curvefs: Merge updateInode and GetOrModifyS3ChunkInfo 2 rpcs in backg…

### PR DESCRIPTION
**Pull request by baijiaruo**
…round flush process

Signed-off-by: baijiaruo <baijiaruo@126.com>

### What problem does this PR solve?

Issue Number: #1361  

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Review by ilixiaocui — APPROVED** [review]

**Review by opencurveadmin — APPROVED** [review]

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

File: curvefs/src/metaserver/inode_storage.cpp [hunk 1]
```diff
@@ -410,16 +418,29 @@ MetaStatusCode InodeStorage::ModifyInodeS3ChunkInfoList(
     const S3ChunkInfoList* list2del) {
     WriteLockGuard lg(rwLock_);
     auto txn = kvStorage_->BeginTransaction();
+    uint32_t step = 0;
```
Comment (Wine93) [inline curvefs/src/metaserver/inode_storage.cpp:421]: you can use a meaningful const variables to replace `step` variable.

File: curvefs/src/metaserver/inode_storage.cpp [hunk 2]
```diff
@@ -49,7 +49,7 @@ using ::curvefs::metaserver::storage::Prefix4InodeS3ChunkInfoList;
 using ::curvefs::metaserver::storage::Prefix4AllInode;
 using ::curvefs::metaserver::storage::Key4InodeAuxInfo;
 
-using Transaction = std::shared_ptr<StorageTransaction>;
+// using Transaction = std::shared_ptr<StorageTransaction>;
```
Comment (Wine93) [inline curvefs/src/metaserver/inode_storage.cpp:52]: delete it directly.

File: curvefs/src/metaserver/inode_manager.cpp [hunk 3]
```diff
@@ -316,6 +316,24 @@ MetaStatusCode InodeManager::UpdateInode(const UpdateInodeRequest& request,
         ++(*deletedNum);
     }
 
+    const S3ChunkInfoMap &map2add = request.s3chunkinfoadd();
```
Comment (Wine93) [inline curvefs/src/metaserver/inode_manager.cpp:319]: why add this map, and it not exist in before?

Comment (baijiaruo) [reply curvefs/src/metaserver/inode_manager.cpp:319]: You can look at the GetOrModifyS3ChunkInfoRequest interface to request the map that has been used all the time, here it is just merged into UpdateInodeRequest

File: curvefs/src/metaserver/inode_storage.cpp (hunk 2 above)
Comment (baijiaruo) [reply curvefs/src/metaserver/inode_storage.cpp:52]: fixed

File: curvefs/src/metaserver/inode_storage.cpp [hunk 4]
```diff
@@ -410,16 +418,29 @@ MetaStatusCode InodeStorage::ModifyInodeS3ChunkInfoList(
...
         rc = AddS3ChunkInfoList(txn, fsId, inodeId, chunkIndex, list2add);
+        step++;
+    }
+
+    if (rc == MetaStatusCode::OK) {
+        uint64_t size4add =
+            (nullptr == list2add) ? 0 : list2add->s3chunks_size();
+        uint64_t size4del =
+            (nullptr == list2del) ? 0 : list2del->s3chunks_size();
+        rc = UpdateInodeS3MetaSize(txn, fsId, inodeId, size4add, size4del);
```
Comment (Wine93) [inline curvefs/src/metaserver/inode_storage.cpp:438]: please add TODO here.

File: curvefs/src/metaserver/inode_storage.cpp (hunk 1 above)
Comment (baijiaruo) [reply curvefs/src/metaserver/inode_storage.cpp:421]: fixed

File: curvefs/src/metaserver/inode_storage.cpp (hunk 4 above)
Comment (baijiaruo) [reply curvefs/src/metaserver/inode_storage.cpp:438]: added

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Comment by baijiaruo** [comment]
recheck

**Review by xu-chaojie — APPROVED** [review]

**Pull request merged**